pkg/sass-parser/lib/src/expression/number.ts

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as postcss from 'postcss';
+
+import {LazySource} from '../lazy-source';
+import type * as sassInternal from '../sass-internal';
+import * as utils from '../utils';
+import {Expression} from '.';
+
+/**
+ * The initializer properties for {@link NumberExpression}.
+ *
+ * @category Expression
+ */
+export interface NumberExpressionProps {
+  value: number;
+  unit?: string;
+  raws?: NumberExpressionRaws;
+}
+
+/**
+ * Raws indicating how to precisely serialize a {@link NumberExpression}.
+ *
+ * @category Expression
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface -- No raws for a number expression yet.
+export interface NumberExpressionRaws {}
```

Comment:
For both Booleam and Number.. I don't think we have raws, unless we want to have a `before`. But I based this on `StringExpression` which doesn't. I left them empty in here to be consistent, but it smells of YAGNI.

Should I remove these if they are empty?

Reply:
For `NumberExpression`, we should have a `{raw: string, value: string}`-style raw for the number's value, because the same parsed number can be represented in different ways. For example, `1` can be represented as `1.0`, `01`, and so on.

For `BooleanExpression`, as I mentioned in chat, we may want to add general support for `ExpressionRaws.before`. Let's keep this in place for now, and we can remove it later if we decide it won't be useful.

Reply:
As discussed in chat, I'll leave the `before`/`after` for all expressions as out of scope for this PR, and I only added the number raw value.

That said, I wasn't sure why is `{raw: string, value: string}` needed. Can this use `{value: string}` only as Raws? or what case do you have in mind where `raw` and `value` (in `raws`) would have different values?

Reply:
Sorry, for a number it should be `{raw: string, value: number}`. The point is to only apply the raw string if the value matches the actual value of the expression. For example, suppose you have a PostCSS plugin that rounds all numbers to 3 decimal places. The number expression would be

```js
{
  sassType: "number",
  value: 0.1234567,
  raws: {raw: ".1234567", value: 0.1234567}
}
```

Since this plugin doesn't care about formatting, it doesn't bother setting raws. It just sets the `number.value = 0.123`. The serializer then checks whether `number.raws.value === number.value`, and if it doesn't it ignores the raw. This way, it avoids overwriting the modified value with the original raw.